201-api-management-logs-to-moesif-using-eventhub-webapp/azuredeploy.json

```diff
@@ -0,0 +1,324 @@
+                    },
+                    "tags": {
+                        "value": "[variables('tags')]"
+                    }
+                }
+            }
+        },
+        {
+            "type": "Microsoft.Resources/deployments",
+            "apiVersion": "2019-10-01",
+            "name": "eventhub-deploy",
+            "properties": {
+                "mode": "Incremental",
+                "templateLink": {
+                    "uri": "[variables('azEventHubTemplUri')]",
+                    "contentVersion": "1.0.0.0"
+                },
+                "parameters": {
+                    "eventHubNsName": {
+                        "value": "[parameters('dnsNamePrefix')]"
+                    },
+                    "eventHubName": {
+                        "value": "[parameters('dnsNamePrefix')]"
+                    },
+                    "eventHubSendPolicyName": {
+                        "value": "[variables('eventHubSendPolicyName')]"
+                    },
+                    "eventHubListenPolicy": {
+                        "value": "[variables('eventHubListenPolicy')]"
+                    },
+                    "tags": {
+                        "value": "[variables('tags')]"
+                    }
+                }
+            }
+        },
+        {
+            "type": "Microsoft.Resources/deployments",
+            "apiVersion": "2019-10-01",
+            "name": "api-management-logger-res-grp-deploy",
+            "condition": "[variables('apiMgrSpecified')]",
+            "resourceGroup": "[parameters('existingApiMgmtResourceGroup')]",
+            "dependsOn": [
+                "[resourceId('Microsoft.Resources/deployments', 'eventhub-deploy')]"
+            ],
+            "properties": {
+                "mode": "Incremental",
+                "expressionEvaluationOptions": {
+                    "scope": "inner"
+                },
+                "parameters": {
+                    "existingApiMgmtName": {
+                        "value": "[parameters('existingApiMgmtName')]"
+                    },
+                    "logToEventhubLoggerId": {
+                        "value": "[variables('apiManagementLoggerId')]"
+                    },
+                    "eventHubNS": {
+                        "value": "[parameters('dnsNamePrefix')]"
+                    },
+                    "eventHubName": {
+                        "value": "[parameters('dnsNamePrefix')]"
+                    },
+                    "eventHubPolicyName": {
+                        "value": "[variables('eventHubSendPolicyName')]"
+                    },
+                    "tags": {
+                        "value": "[variables('tags')]"
+                    },
+                    "templLoggerUrl": {
+                        "value": "[variables('azApiMgmtLoggerInstallTemplUri')]"
+                    }
+                },
+                "template": {
```

Comment:
move this to a file/templatelink to be consistent with the rest

Reply:
I spent a fair amount of time today on moving the nested deployment to a separate file with templateLink url.
I kept getting this error as I copy pasted the azuredeploy.json into azure portal: `The provided content link 'Microsoft.WindowsAzure.ResourceStack.Frontdoor.Data.Entities.Deployments.DeploymentTemplateContentLink' is invalid or not supported. Content link must be an absolute URI not referencing local host or UNC path. (Code: InvalidContentLink)` 

For `templateLink/url` what worked is using actual full github URL.
I tried the following and they all failed with similar error code above:
```
[concat('<hardcode base uri>', 'nested/<new-json-template>.json')]
[uri(parameters('_artifactsLocation'), 'nested/<new-json-template>.json')]
[concat(parameters('_artifactsLocation'), 'nested/<new-json-template>.json')]
[concat(parameters('_artifactsLocation'), 'nested/<new-json-template>.json', parameters('_artifactsLocationSasToken'))]
[uri(parameters('_artifactsLocation'), concat('nested/<new-json-template>.json', parameters('_artifactsLocationSasToken')))]

```
So I eventually gave up. If this is important, and have helpful pointers that would be great.

Reply:
The last one is what you want (same pattern as the rest of the linked templates you have) - if everything else is working and that's failing, just submit the PR and we can debug.

Reply:
To simplify things, decided to initially restrict existing API Management to be in same resource group as rest of resources being deployed. After this, will evaluate cross resource group. The two main issues I faced were the `DeploymentTemplateContentLink` mentioned above, as well as `dependsOn` not working when obtaining connection string in nested templates - though [your Blog Post on this article](https://bmoore-msft.blog/2020/07/26/resource-not-found-dependson-is-not-working/) was helpful - but it recently failed again. So I decided to take incremental approach. 

BTW: If you have link/blog to `apiVersion` to use, please let me know. The Run Best Practices pipeline check fails because some apiVersion were over 2 years old. However, VS Code Intellisense (autosuggest) shows these were latest published API versions.

Reply:
VS Code is stale - the correct/latest versions are listed in the test results, so please update to use those.

Reply:
Done. Thanks for clarifying. Updated api versions. All tests are green right now.